src/pymovements/dataset/resources.py

```diff
@@ -0,0 +1,234 @@
+
+from collections.abc import Iterable
+from collections.abc import Sequence
+from copy import deepcopy
+from dataclasses import asdict
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
+class ResourceDefinition:
+    """ResourceDefinition definition."""
+
+    content: str
+    filename: str | None = None
+    url: str | None = None
+    md5: str | None = None
+
+    @staticmethod
+    def from_dict(dictionary: dict[str, Any]) -> ResourceDefinition:
+        """Create a ``Resource`` instance from a dictionary.
+
+        Parameters
+        ----------
+        dictionary : dict[str, Any]
+            A dictionary containing Resource parameters.
+
+        Returns
+        -------
+        ResourceDefinition
+            An initialized ``Resource`` instance.
+        """
+        if 'resource' in dictionary:
+            url = dictionary['resource']
+            dictionary = {key: value for key, value in dictionary.items() if key != 'resource'}
+            dictionary['url'] = url
+
+        return ResourceDefinition(**dictionary)
+
+    def to_dict(self, *, exclude_none: bool = True) -> dict[str, Any]:
+        """Convert the ``ResourceDefinition`` instance into a dictionary.
+
+        Parameters
+        ----------
+        exclude_none: bool
+            Exclude attributes that are either ``None`` or that are objects that evaluate to
+            ``False`` (e.g., ``[]``, ``{}``). Attributes of type ``bool``, ``int``, and ``float``
+            are not excluded.
+
+        Returns
+        -------
+        dict[str, Any]
+            ``dict`` representation of ``ResourceDefinition``.
+        """
+        data = asdict(self)
+
+        # Delete fields that evaluate to False (False, None, [], {})
+        if exclude_none:
+            for key, value in list(data.items()):
+                if not isinstance(value, (bool, int, float)) and not value:
+                    del data[key]
+
+        return data
+
+
+class ResourceDefinitions(list):
+    """List of ``ResourceDefinition`` instances."""
+
+    def __init__(self, resources: Iterable[ResourceDefinition] | None = None) -> None:
+        if resources is None:
+            super().__init__([])
+        else:
+            super().__init__(resources)
+
+    def filter(self, content: str | None = None) -> ResourceDefinitions:
+        """Filter ``ResourceDefinitions`` for content type.
+
+        Parameters
+        ----------
+        content: str | None
+            The content type to filter for. If ``None``, then don't filter. (default: None)
+
+        Returns
+        -------
+        ResourceDefinitions
+            A new ``ResourceDefinitions`` instance that contains only resources of the specified
+            content type.
+        """
+        if content is None:
+            return self
+
+        resources = [resource for resource in self if resource.content == content]
+        return ResourceDefinitions(resources)
+
+    @staticmethod
+    def from_dict(
```

Comment:
this is the legacy nested dictionary format that will be deprecated in a followup